app/move/app/new/controllers/base.js

```diff
@@ -133,14 +134,36 @@ class CreateBaseController extends FormWizardController {
       id: currentLocationId,
       location_type: locationType,
       can_upload_documents: canUploadDocuments,
+      young_offender_institution: isYoungOffenderInstitution,
     } = req.session.currentLocation || {}
 
     req.form.values.from_location = currentLocationId
     req.form.values.from_location_type = locationType
     req.form.values.can_upload_documents = canUploadDocuments
+    req.form.values.is_young_offender_institution = isYoungOffenderInstitution
 
     super.saveValues(req, res, next)
   }
+
+  shouldAskYouthSentenceStep(req, res, next) {
+    const fromLocationType = req.sessionModel.get('from_location_type')
+    const isYoungOffenderInstitution = req.sessionModel.get(
+      'is_young_offender_institution'
+    )
+    const { date_of_birth: dateOfBirth } = req.sessionModel.get('person') || {}
+    const age = filters.calculateAge(dateOfBirth)
+
+    if (
+      fromLocationType === 'prison' &&
+      isYoungOffenderInstitution &&
+      age >= 18 &&
```

Comment:
Would this make more sense if the age ranges lived in the config. (Same goes for `app/moves/middleware/set-body-request-filters/age.js` - and anywhere else???)

Reply:
The request filters are a separate thing I feel.

But in terms of this, yeah maybe it might be better to define this in the config to be more specific.